server.go

```diff
@@ -596,6 +601,42 @@ func (s *Server) inheritClientSession(pk packets.Packet, cl *Client) bool {
 		return true // [MQTT-3.2.2-3]
 	}
 
+	// Look up a stored client that's not in memory yet:
+	if s.hooks.Provides(StoredClientByID) {
```

Comment:
Same issue here. Possible solution would be to add this logic to the OnSessionEstablish hook and overwrite the values of Client.State directly, but without adding the client to the server. This will avoid the client being picked up in inheritClientSession and overwriting the new networked client values.